docs/RFCS/20171214_computed_columns.md

```diff
@@ -0,0 +1,155 @@
+- Feature Name: Computed Columns
+- Status: draft
+- Start Date: 2017-12-14
+- Authors: Daniel Harrison, Justin Jaffray
+- RFC PR:
+- Cockroach Issue:
+
+# Summary
+
+Computed columns allow users to create columns which are not directly writable,
+but are instead pure functions of other column values in that row. This is
+supported in several major databases, including MySQL, Oracle, and SQL Server.
+This RFC covers both the version of computed columns that are physically stored
+alongside other columns (referred to here as *materialized*) and computed
+columns which are only computed when read (*non-materialized*).
+
+# Motivation
+
+Computed columns allow storing data derived from other columns without putting
+the burden of keeping that data updated on the client. See [Guide Level
+Explanation: Partitioning](#partitioning) and [Guide Level Explanation: JSON](#json) for the two
+motivating examples.
+
+
+# Guide Level Explanation
+
+The grammar for a computed column is `column_name <type> AS <expr>
+[PERSISTED]`, where `<expr>` is a pure function of non-computed columns in the
+same table. `PERSISTED` indicates a materialized computed column and is
```

Comment:
Do we really need PERSISTED ?I ask because it appears that we're only using it for indexes.